### PR TITLE
add section for orchestrating resource deployment

### PR DESCRIPTION
**Pull request by cbodonnell**
[SC-77370](https://app.shortcut.com/replicated/story/77370/implement-kots-wait-for-ready-feature-for-native-kubernetes-manifests)

Supporting documentation for https://github.com/replicatedhq/kots/pull/3903

### CODE REVIEW

File: docs/vendor/orchestrating-resource-deployment.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+#### `kots.io/creation-phase: '<integer>'`
+
+When this annotation is present on a resource, KOTS groups the resource into the specified creation phase. KOTS deploys each phase in order from lowest to highest. Resources within the same phase are deployed in the same order that Helm installs resources. For more information, see [Using Helm](https://helm.sh/docs/intro/using_helm/#:~:text=Helm%20installs%20resources%20in%20the,order) in the Helm documentation.
+
+#### `kots.io/deletion-phase: '<integer>'`
+
+When this annotation is present on a resource, KOTS groups the resource into the specified deletion phase. KOTS deletes each phase in order from lowest to highest. Resources within the same phase are deleted in the reverse order from which they were created.
+
+:::note
+Kubernetes annotations cannot be integers and must be strings, so make sure to quote this.
```
Comment (ajp-io) [inline docs/vendor/orchestrating-resource-deployment.md:24]: "quote this" feels a bit unspecific. @jonquil2002 what about this?

Also this note applies to the creation and deletion phases. Do you think the note is in the right place, or does it look like this only applies to the deletion phase?
```suggestion
Kubernetes annotations cannot be integers and must be strings, so make sure to quote the integer in the annotation.
```

File: docs/vendor/orchestrating-resource-deployment.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+```
+
+### Wait for a Resource to be Ready
+
+#### `kots.io/wait-for-ready: '<bool>'`
+
+When this annotation is present on a resource and evaluates to `'true'`, KOTS waits for the resource to be in a Ready state before deploying any other resources. This leverages the same logic as application status informers to determine if a resource is ready. If there is no existing status informer for a given resource type, KOTS waits until the resource exists and is queryable from the Kubernetes API server. For more information about status informers, see [Displaying Application Status](/vendor/admin-console-display-app-status#about-status-informers).
+
+:::note
+Kubernetes annotations cannot be booleans and must be strings, so make sure to quote this.
```
Comment (ajp-io) [inline docs/vendor/orchestrating-resource-deployment.md:49]: Same "quote this" comment, unless you feel like it's sufficiently clear.

File: docs/vendor/orchestrating-resource-deployment.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+:::
+
+#### Example
+
+The following example causes KOTS to wait for the postgres `StatefulSet` to be ready before continuing to deploy other resources.
```
Comment (ajp-io) [inline docs/vendor/orchestrating-resource-deployment.md:54]:
```suggestion
The following example causes KOTS to wait for the Postgres `StatefulSet` to be ready before continuing to deploy other resources.
```

File: docs/vendor/orchestrating-resource-deployment.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
+# Orchestrating Resource Deployment
```
Comment (ajp-io) [inline docs/vendor/orchestrating-resource-deployment.md:1]: We should add a link from the release notes to this page.

**Comment by crdant** [comment]
I learned this morning that the release went out including this feature and was looking to share the docs. I think the "HOLD FOR RELEASE" no longer applies here.

**Comment by ajp-io** [comment]
It doesn't, we just need @jonquil2002 to do a review before we can merge it.

**Review by paigecalvert — DISMISSED** [review]

**Review by paigecalvert — APPROVED** [review]

**Pull request merged**